ui/tests/integration/components/lifecycle-chart-test.js

```diff
@@ -90,16 +126,69 @@ module('Integration | Component | lifecycle-chart', function(hooks) {
       assert.notOk(task.isFinished);
     });
 
-    this.set('taskStates.firstObject.state', 'running');
+    // Change poststart-ephemeral to be running
+    this.set('taskStates.4.state', 'running');
     await settled();
 
-    assert.ok(Chart.phases[1].isActive);
-    assert.ok(Chart.tasks[3].isActive);
     await componentA11yAudit(this.element, assert);
 
-    this.set('taskStates.firstObject.finishedAt', new Date());
+    assert.ok(Chart.tasks[5].isActive);
+
+    assert.ok(Chart.phases[1].isActive);
+    assert.notOk(
+      Chart.phases[2].isActive,
+      'the poststart phase is nested within main and should never have the active class'
+    );
+
+    this.set('taskStates.4.finishedAt', new Date());
     await settled();
 
-    assert.ok(Chart.tasks[3].isFinished);
+    assert.ok(Chart.tasks[5].isFinished);
+  });
+
+  [
+    {
+      testName: 'expected active phases',
+      runningTaskNames: ['prestart ephemeral', 'main one', 'poststop'],
+      activePhaseNames: ['Prestart', 'Main', 'Poststop'],
+    },
+    {
+      testName: 'sidecar task states don’t affect phase active states',
+      runningTaskNames: ['prestart sidecar', 'poststart sidecar'],
+      activePhaseNames: [],
+    },
+    {
+      testName: 'poststart ephemeral task states affect main phase active state',
+      runningTaskNames: ['poststart ephemeral'],
+      activePhaseNames: ['Main'],
+    },
+  ].forEach(async ({ testName, runningTaskNames, activePhaseNames }) => {
+    test(testName, async function(assert) {
+      this.set(
+        'taskStates',
+        tasks.map(task => {
+          return { task };
+        })
```

Comment:
Mega tiny thing but if you write this as `tasks.map(task => ({ task }))` Prettier might not break it into six lines.

Reply:
hah true, done, thanks 😀